lib/validator.js

```diff
@@ -17,37 +23,65 @@ var inMapping = {
  * @param parameters list of params
  * @returns JSON schema
  */
-function convertToJsonSchema(parameters) {
+Validator.prototype._convertToJsonSchema = function(parameters) {
     var schema = {
         type: 'object',
         properties: {}
     };
 
     parameters.forEach(function(param) {
-        if (!schema.properties[inMapping[param.in]]) {
-            schema.properties[inMapping[param.in]] = {
-                type: 'object',
-                properties: {}
-            };
-            // 'required' array must have at least one element according to json-schema spec,
-            // se we can't preinitialize it.
-            schema.required = schema.required || [];
-            schema.required.push(inMapping[param.in]);
-        }
-        var reqPartSchema = schema.properties[inMapping[param.in]];
-        reqPartSchema.properties[param.name] = {};
-        if (param.required) {
-            reqPartSchema.required = reqPartSchema.required || [];
-            reqPartSchema.required.push(param.name);
+        if (param.in !== 'body') {
+            if (!schema.properties[inMapping[param.in]]) {
+                schema.properties[inMapping[param.in]] = {
+                    type: 'object',
+                    properties: {}
+                };
+                // 'required' array must have at least one element according to json-schema spec,
+                // se we can't preinitialize it.
+                schema.required = schema.required || [];
+                schema.required.push(inMapping[param.in]);
+            }
+
+            var reqPartSchema = schema.properties[inMapping[param.in]];
+            var paramSchema = {};
+            // We can't type-check directly, because everything come in as a string
```

Comment:
@gwicke I've tried that approach, but, for `path` parameters it breaks routing, and I didn't really investigate further. We could still try that, either making `path` params an exception, or adding a `toString()` call in swagger-router somewhere around [here](https://github.com/wikimedia/swagger-router/blob/master/lib/router.js#L103). But, may be we'll have more implications - I'll check tomorrow.